src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java

```diff
@@ -332,4 +332,74 @@ private boolean isMatchingByLineAndColumnAndTokenType(DetailAST ast) {
                 && expandedTabColumn(ast) == columnNumber
                 && (tokenType == 0 || tokenType == ast.getType());
     }
+
+    /**
+     * Escape &lt;, &gt;, &amp;, &#39; and &quot; as their entities.
+     * Custom method for Xpath generation to maintain compatibility
+     * with Saxon and encoding outside Ascii range characters.
+     *
+     * @param value the value to escape.
+     * @return the escaped value if necessary.
+     */
+    public static String encode(String value) {
+        final StringBuilder sb = new StringBuilder(256);
+        value.chars().forEach(
```

Comment:
instead of chars please use "codePoints()" to work with unicode.
Please make Input + test method that use unicode and it would fail if we use `chars()` here

Reply:
Sir, can you please suggest how do I convert an Integer (ASCII value of char given by codepoints) to char without cast, as cast is not allowed by TeamCity.

Reply:
you do `encodeCharacter(Character.toChars(chr)[0])` now .. why `[0]` ?

Please create Input file with String value like `"语言处理"` and how it can be escaped/suppressed by xpath.

Reply:
`[0]` is used because `.toChars(chr)` returns a character array and we require only the first element of the array. Alright, I will add those examples.

Reply:
please show by UT that encoding of unicode symbols continue to work fine.
plase add unicoded strings to  `public void testEncode() {`

Reply:
Done.

Reply:
please add unicode in string constants - it is allowed.
as very very last resort please add suppression for this file+check at https://github.com/checkstyle/checkstyle/blob/master/config/suppressions.xml

Reply:
Done, No more required as they are not being encoded.